app/invocation/invocation_artifacts_card.tsx

```diff
@@ -35,50 +34,78 @@ export default class ArtifactsCardComponent extends React.Component {
+        outputs: visibleOutputs,
+        hiddenOutputCount: target.outputs.length - visibleOutputs.length,
+      });
+    }
+    const hiddenTargetCount = filteredTargets.length - visibleTargets.length;
 
     return (
       <div className="card artifacts">
         <img className="icon" src="/image/arrow-down-circle.svg" />
         <div className="content">
           <div className="title">Artifacts</div>
           <div className="details">
-            {targets
-              .slice(0, (this.props.pageSize && this.state.numPages * this.props.pageSize) || undefined)
-              .map((completed) => (
-                <div>
-                  <div className="artifact-section-title">{completed.id.targetCompleted.label}</div>
-                  {completed.completed.importantOutput
-                    .filter(
-                      (output) =>
-                        !this.props.filter ||
-                        completed.id.targetCompleted.label.toLowerCase().includes(this.props.filter.toLowerCase()) ||
-                        output.name.toLowerCase().includes(this.props.filter.toLowerCase())
-                    )
-                    .map((output) => (
-                      <a
-                        href={rpcService.getBytestreamFileUrl(output.name, output.uri, this.props.model.getId())}
-                        className="artifact-name"
-                        onClick={this.handleArtifactClicked.bind(this, output.uri, output.name)}>
-                        {output.name}
-                      </a>
-                    ))}
+            {visibleTargets.map((target) => (
+              <div>
+                <div className="artifact-section-title">{target.label}</div>
+                <div className="artifact-list">
+                  {target.outputs.map((output) => (
+                    <a
+                      href={rpcService.getBytestreamFileUrl(output.name, output.uri, this.props.model.getId())}
+                      className="artifact-name"
+                      onClick={this.handleArtifactClicked.bind(this, output.uri, output.name)}>
+                      {output.name}
+                    </a>
+                  ))}
+                  {target.hiddenOutputCount > 0 && (
+                    <div className="artifact-hidden-count">
+                      + {target.hiddenOutputCount} more {target.hiddenOutputCount === 1 ? "artifact" : "artifacts"} for
+                      this target
+                    </div>
+                  )}
                 </div>
-              ))}
-            {targets.flatMap((completed) => completed.completed.importantOutput).length == 0 && (
-              <span>{this.props.filter ? "No matching artifacts" : "No artifacts"}</span>
+              </div>
+            ))}
+            {hiddenTargetCount > 0 && (
+              <div className="artifact-hidden-count">
+                + {hiddenTargetCount} more {hiddenTargetCount === 1 ? "target" : "targets"} with artifacts
```

Comment:
Should clicking on this (or the one below) load more artifacts?

Reply:
Since it's hard to say what the most intuitive thing to do is when clicking those, I just made them less "clickable" looking per our offline discussion (see the updated screenshot).